stl/inc/xmemory

```diff
@@ -2003,6 +2007,99 @@ typename _Container::size_type _Erase_nodes_if(_Container& _Cont, _Pr _Pred) {
     }
     return _Old_size - _Cont.size();
 }
+
+#if _HAS_CXX20
+template <class _Ty, class _Alloc, class... _Types, enable_if_t<!_Is_specialization_v<_Ty, pair>, int> = 0>
+constexpr auto uses_allocator_construction_args(const _Alloc& _Al, _Types&&... _Args) noexcept {

```

Comment:
Should some of these functions be `_NODISCARD`? It appears that all of the `uses_allocator_construction_args()` overloads, and `make_obj_using_allocator()`, should be `_NODISCARD`, as their only purpose is to generate their return values. However, `uninitialized_construct_using_allocator()` has side effects and the return value isn't essential, so it shouldn't be marked.

(I could be wrong here as I am not caffeinated :joy_cat:)

Reply:
Done